tests/functional/object/mcg/test_s3_vector.py

```
@@ -1310,3 +1331,592 @@ def test_vector_distance_metrics(self, vector_bucket_factory, mcg_obj):
+                )
+            else:
+                logger.info(
+                    "✓ Policy is empty after deletion — default access rules apply"
+                )
+
+        except ClientError as e:
+            # Step 13: Policy-not-found error is the expected NooBaa/S3 response
+            error_code = e.response.get("Error", {}).get("Code", "")
+            assert (
+                error_code == "NoSuchVectorBucketPolicy"
+            ), f"Unexpected error after policy deletion: {error_code} — {e}"
```

Ruff (0.15.18)

[warning] Found assertion on exception `e` in `except` block, use `pytest.raises()` instead

(PT017)